runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecord.java

```diff
@@ -43,81 +41,105 @@
     @Getter
     private ISerializer serializerType;
 
-    /** An undo record, which can be used to undo this method.
-     *
+    /**
+     * An undo record, which can be used to undo this method.
      */
     @Getter
     public transient Object undoRecord;
 
-    /** A flag indicating whether an undo record is present. Necessary
+    /**
+     * A flag indicating whether an undo record is present. Necessary
      * because undo records may be NULL.
      */
     @Getter
     public boolean undoable;
 
-    /** The upcall result, if present. */
+    /**
+     * The upcall result, if present.
+     */
     @Getter
     public transient Object upcallResult;
 
-    /** If there is an upcall result for this modification. */
+    /**
+     * If there is an upcall result for this modification.
+     */
     @Getter
     public transient boolean haveUpcallResult = false;
 
-    /** Set the upcall result for this entry. */
+    /**
+     * The global address this record is associated to.
+     */
+    @Getter
+    @Setter
+    long globalAddress = Address.NON_ADDRESS;
```

Comment:
Thinking about this again, it should have an extra cost, since storing a parent reference would be the same memory overhead.

Reply:
Yes, that previous patch was not intended to save memory, but to remove circular dependency, making it more readable and less error prone.